### PR TITLE
Add --cni and --cloud-provider-name=vsphere support

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

* Add support for `--cni` flag, special behavior for `--cloud-provider-name=vsphere` on servers
* Provide separate image tarballs for each supported CNI, and for vSphere
* Pass through critical configuration values to helm charts via --set (default registry, CIDRs, DNS config, etc)
* Add new flags to agent/server config docs; add additional image tarballs to airgap docs

#### Types of Changes ####

* CLI
* Packaged components

#### Verification ####

* Start RKE2 no extra flags; validate that normal function is observed
* Start RKE2 with --cni=none; validate that no CNI is deployed
* Start RKE2 with --cloud-provider-name=vsphere; validate that vsphere charts are deployed (TODO - need vSphere lab access for QA to test this)

#### Linked Issues ####

CNI flag: #887
vSphere CPI/CSI: #916 
Chart values: #917 (requires rancher/rke2-charts#73)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -170,6 +174,9 @@ COPY --from=charts \
     /charts/ \
     /charts/
 
+FROM scratch AS runtime
+COPY --from=runtime-collect / /
```
Comment (brandond) [inline Dockerfile:178]: Flattening all this into a single layer makes the runtime extraction a couple seconds faster

File: pkg/podexecutor/staticpod.go [hunk 2]
```diff
@@ -57,6 +57,8 @@ type CloudProviderConfig struct {
 func (s *StaticPodConfig) Kubelet(args []string) error {
 	extraArgs := []string{
 		"--volume-plugin-dir=/var/lib/kubelet/volumeplugins",
+		"--file-check-frequency=5s",
+		"--sync-frequency=30s",
```
Comment (brandond) [inline pkg/podexecutor/staticpod.go:61]: Seems to shorten the initial static pod startup by a fair bit, without much additional overhead to steady-state operation.

File: pkg/util/strings.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package util
```
Comment (ibuildthecloud) [inline pkg/util/strings.go:1]: This method almost verbatim is in wrangler already if you want to use it: https://github.com/rancher/wrangler/blob/master/pkg/slice/contains.go#L3

File: pkg/rke2/rke2.go [hunk 4]
```diff
@@ -26,15 +26,15 @@ import (
 )
 
 type Config struct {
-	CloudProviderName   string
-	CloudProviderConfig string
 	AuditPolicyFile     string
-	KubeletPath         string
+	CNI                 string
```
Comment (ibuildthecloud) [inline pkg/rke2/rke2.go:30]: Are these fields used anywhere

File: pkg/cli/cmds/server.go [hunk 5]
```diff
@@ -1,19 +1,33 @@
...
-	config rke2.Config
+	DisableItems = []string{"rke2-coredns", "rke2-ingress-nginx", "rke2-kube-proxy", "rke2-metrics-server"}
+	CNIItems     = []string{"canal", "cilium"}
+
+	config = rke2.Config{}
+
+	serverFlag = []cli.Flag{
+		&cli.StringFlag{
+			Name:   "cni",
+			Usage:  "(networking) CNI Plugin to deploy, one of none, " + strings.Join(CNIItems, ", "),
```
Comment (manuelbuil) [inline pkg/cli/cmds/server.go:26]: one of none?

**Review by manuelbuil — COMMENTED** [review]
LGTM

File: pkg/cli/cmds/server.go (hunk 5 above)
Comment (brandond) [reply pkg/cli/cmds/server.go:26]: This is joined with the CNI items list, and uses a common phrasing for selecting one value from a list of valid items:
`one of none, canal, calico`

**Review by ibuildthecloud — APPROVED** [review]

File: pkg/podexecutor/staticpod.go (hunk 2 above)
Comment (Oats87) [reply pkg/podexecutor/staticpod.go:61]: Tuned by Rancher!

**Review by Oats87 — APPROVED** [review]

**Review by dweomer — APPROVED** [review]

**Pull request merged**